input/sdl_gamepad.c

```diff
@@ -160,6 +161,13 @@ static void read_gamepad_thread(struct mp_input_src *src, void *param)
 
     SDL_GameController *controller;
     struct priv *p = talloc_zero(src, struct priv);
+    p->cancel_wakeup = SDL_RegisterEvents(1);
```

Comment:
Unfortunately, SDL_RegisterEvents is a global thing. Events can't be unregistered, so every initialization of the gamepad code will consume one more global event. I think you don't need to care about this. If you do, use pthread_once to register a single global event.